apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -77,6 +77,7 @@ export class UpsertWorkflowUseCase {
   async execute(command: UpsertWorkflowCommand): Promise<WorkflowResponseDto> {
     const workflowForUpdate = await this.queryWorkflow(command);
     const persistedWorkflow = await this.createOrUpdateWorkflow(workflowForUpdate, command);
+    // TODO: this upsertControlValues logic should be moved to the create/update workflow usecase
```

Comment:
IMO the logic to create/update the control values should be moved to the create/update workflow usecase, now it's disconnected from the place where we manage steps.

Reply:
In that case, won't we duplicate much of the control value logic? I guess we can get away with a shared use case about control values.

Reply:
I'm not sure why you are saying that the logic will be duplicated. This is the only place where we create/update the control values, all I'm saying is to move this logic to the corresponding workflow usecase.